### ISSUE TITLE
cINN NLL loss computation

### ISSUE THREAD

**Issue by zuoshifan** [jleuschn/cinn_for_imaging#1]
Hi, the cINN NLL loss computed in cinn_for_imaging/cinn_for_imaging/util/torch_losses.py line 139 or line 168 seems not correct.

For a cINN model with a normal distribution, the NLL loss = ||z||_2^2 / 2- log(jac), the correct NLL loss should be something like
`torch.mean(torch.norm(zz, dim=1)**2 / 2 - log_jac)`

For a cINN model with a radial distribution, the NLL loss = (n - 1)*log(||z||_2) + ||z||_2^2 / 2- log(jac), the correct NLL loss should be something like
`torch.mean((ndim_total - 1) * torch.log(torch.norm(zz, dim=1)) + torch.norm(zz, dim=1)**2 / 2 - log_jac)`

**Comment by jleuschn** [comment]
Hi,
thanks for commenting. I think both are equal up to division by `ndim_total` (a constant scaling).

For the normal distribution:
```
yours = torch.mean(torch.norm(zz, dim=1)**2 / 2 - log_jac)
currently_in_code = torch.mean(zz**2) / 2 - torch.mean(log_jac) / ndim_total  #  == yours / ndim_total
```

This is due to
```
torch.norm(zz, dim=1)**2 / ndim_total
== torch.sum(zz**2, dim=1) / ndim_total
== torch.mean(zz**2, dim=1)
```
The averaging over dimension `1` is contained in the `torch.mean(zz**2)` in the current code.

For the radial distribution it appears to be analogously.

The constant term `c` in the normal distribution loss indeed seems wrong to me, but that should not change the gradient used for optimization.

**Comment by zuoshifan** [comment]
Thanks, I understand now.
